internal/http/client.go

```diff
@@ -17,7 +18,13 @@ func NewRetryableClient(timeout time.Duration) *retryablehttp.Client {
 		RetryWaitMin: 1 * time.Second,
 		RetryWaitMax: 30 * time.Second,
 		RetryMax:     3,
-		CheckRetry:   retryablehttp.DefaultRetryPolicy,
+		CheckRetry: func(ctx context.Context, resp *http.Response, err error) (bool, error) {
+			ok, e := retryablehttp.DefaultRetryPolicy(ctx, resp, err)
+			if !ok && resp.StatusCode == http.StatusNotFound {
+				return true, nil
+			}
```

Comment:
I think it'd be better to do an error check first, because `DefaultRetryPolicy()` checks for context errors, which could be user interrupts for instance, in which case we don't want to retry under any circumstance:
```suggestion
			ok, e := retryablehttp.DefaultRetryPolicy(ctx, resp, err)
			if e != nil {
				return ok, e
			}
			if !ok && resp.StatusCode == http.StatusNotFound {
				return true, nil
			}
```